src/api/CoinBase.ts

```diff
@@ -0,0 +1,26 @@
+import { CoinBaseResponse, SourceApi } from '../types/RelayPricer';
+import fetch, { Response } from 'node-fetch';
+
+export default class CoinBase implements SourceApi {
+    // we should get this address from config file or enviroments
+    private readonly URL: string = 'https://api.coinbase.com/v2/exchange-rates';
+
+    async query(input: string, output: string): Promise<number> {
+        const upperCaseOutput = output.toUpperCase();
+        const response: Response = await fetch(
+            `${this.URL}?currency=${input}`,
+            {
+                method: 'get'
+            }
+        );
+
+        if (response.status === 200) {
```

Comment:
you can use `response.ok`

Reply:
also I'd define this the other way around so that you have the return value as a green path, i.e, if not ok throw, otherwise continue

Reply:
updated